rskj-core/src/test/java/co/rsk/pcc/AltBN128Test.java

```diff
@@ -617,26 +659,66 @@ private void altBN128OperationTest(String inputString, String expectedOutput, St
         assertThat(errorMessage, ByteUtil.toHexString(output), is(expectedOutput));
 
-        runAgainWithJavaImpl(expectedOutput, input, contractAddress);
+        runAgainWithJavaImpl(expectedOutput, input, contractAddress, expectedRes);
     }
 
-    private void runAgainWithJavaImpl(String expectedOutput, byte[] input, DataWord contractAddress) {
-        if (Utils.isLinux()){
-            // If we are in Linux, we test the Java Implementation
-            AbstractAltBN128 altBN128 = new JavaAltBN128();
+    private void altBN128OperationTestAfterRSKIP197(String inputString, String expectedOutput, String errorMessage,
+                                       DataWord contractAddress, long gasCost, int expectedRes) throws VMException {
+        byte[] input = Hex.decode(inputString);
+        runAgainWithJavaRSKIP197Impl(expectedOutput, input, contractAddress, expectedRes);
+        runAgainWithGoImpl(expectedOutput, input, contractAddress, expectedRes);
+    }
+
+    private void runAgainWithGoImpl(String expectedOutput, byte[] input, DataWord contractAddress, int expectedRes) {
+        if(Utils.isLinux()) {
+            int result;
+            AbstractAltBN128 altBN128 = new GoAltBN128();
 
             if (contractAddress.equals(PrecompiledContracts.ALT_BN_128_ADD_DW)) {
-                altBN128.add(input,input.length);
+                result = altBN128.add(input, input.length);
             } else if (contractAddress.equals(PrecompiledContracts.ALT_BN_128_MUL_DW)) {
-                altBN128.mul(input,input.length);
+                result = altBN128.mul(input, input.length);
             } else {
-                altBN128.pairing(input,input.length);
+                result = altBN128.pairing(input, input.length);
             }
             assertThat(ByteUtil.toHexString(altBN128.getOutput()), is(expectedOutput));
+            assertEquals(expectedRes, result);
+        }
+    }
+
+    private void runAgainWithJavaImpl(String expectedOutput, byte[] input, DataWord contractAddress, int expectedRes) {
+        // If we are in Linux, we test the Java Implementation
+        AbstractAltBN128 altBN128 = new JavaAltBN128();
+
+        int result;
+        if (contractAddress.equals(PrecompiledContracts.ALT_BN_128_ADD_DW)) {
+            result = altBN128.add(input, input.length);
+        } else if (contractAddress.equals(PrecompiledContracts.ALT_BN_128_MUL_DW)) {
+            result = altBN128.mul(input, input.length);
+        } else {
+            result = altBN128.pairing(input, input.length);
         }
+        assertThat(ByteUtil.toHexString(altBN128.getOutput()), is(expectedOutput));
+        assertEquals(expectedRes, result);
     }
 
-    private void altBN128MulTest(BigInteger x1, BigInteger y1, BigInteger scalar, String expectedOutput) throws VMException {
+    private void runAgainWithJavaRSKIP197Impl(String expectedOutput, byte[] input, DataWord contractAddress, int expectedRes) {
```

Comment:
Could we have test calls from compiled Solidity (maybe using the DSL) only to be sure all this new functionality is consumable running test code? Some samples at `https://docs.klaytn.com/smart-contract/precompiled-contracts`

Reply:
Do you think we should add more tests to "PrecompiledContractsCallErrorHandlingTests"? or other kind of tests?

Reply:
I mean, code tests, using explicitly the hash code produced by a Solidity contract or similar tool (the hash of the full signature of event)